### PR TITLE
APL-2848 Handle pipeline id and commit sha to set docker image

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Use the E2E_PIPELINE_ID and E2E_COMMIT_SHA env variables to automatically define the docker image to use for tests using the docker image of the agent

Needs: https://github.com/DataDog/test-infra-definitions/pull/602
Will update go.mod and test-infra image when the PR above is merged

### Motivation

Make things working the same way for the package of the agent and the docker image of the agent

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by KevinFairise2** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=auto

**Comment by KevinFairise2** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=auto

File: test/new-e2e/pkg/runner/configmap.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ const (
 	AgentAPPKey = commonconfig.DDAgentConfigNamespace + ":" + commonconfig.DDAgentAPPKeyParamName
 	// AgentPipelineID pulumi config parameter name
 	AgentPipelineID = commonconfig.DDAgentConfigNamespace + ":" + commonconfig.DDAgentPipelineID
+	// AgentCommitSHA pulumi config parameter name
+	AgentCommitSHA = commonconfig.DDAgentConfigNamespace + ":" + commonconfig.DDAgentCommitSHA
```
Comment (pducolin) [inline test/new-e2e/pkg/runner/configmap.go:27]: **❓ question**
Should we add a warning if `AgentCommitSHA` is set and not `AgentPipelineID` ?

**Review by pducolin — APPROVED** [review]

File: .gitlab/e2e.yml [hunk 2]
```diff
@@ -97,9 +97,10 @@ k8s-e2e-otlp-main:
     E2E_PRIVATE_KEY_PATH: /tmp/agent-qa-ssh-key
     E2E_KEY_PAIR_NAME: datadog-agent-ci-rsa
     E2E_PIPELINE_ID: $CI_PIPELINE_ID
+    E2E_COMMIT_SHA: $CI_COMMIT_SHORT_SHA
     E2E_OUTPUT_DIR: $CI_PROJECT_DIR/e2e-output
   script:
-    - inv -e new-e2e-tests.run --targets $TARGETS -c ddagent:fullImagePath=669783387624.dkr.ecr.us-east-1.amazonaws.com/agent:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA} -c ddagent:clusterAgentFullImagePath=669783387624.dkr.ecr.us-east-1.amazonaws.com/cluster-agent:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA} -c dddogstatsd:fullImagePath=669783387624.dkr.ecr.us-east-1.amazonaws.com/dogstatsd:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA} -c ddagent:imagePullRegistry=669783387624.dkr.ecr.us-east-1.amazonaws.com -c ddagent:imagePullUsername=AWS -c ddagent:imagePullPassword=$(aws ecr get-login-password) --junit-tar "junit-${CI_JOB_NAME}.tgz" ${EXTRA_PARAMS}
+    - inv -e new-e2e-tests.run --targets $TARGETS -c ddagent:imagePullRegistry=669783387624.dkr.ecr.us-east-1.amazonaws.com -c ddagent:imagePullUsername=AWS -c ddagent:imagePullPassword=$(aws ecr get-login-password) --junit-tar "junit-${CI_JOB_NAME}.tgz" ${EXTRA_PARAMS}
```
[the remark this answers is not in the document]
Comment (KevinFairise2) [reply .gitlab/e2e.yml:103]: I think we still need this for now as we need it for the authentication of the docker client to still work. We can probably change things to have it defaulted to the value here when running in the CI. I will create card to adress it in a follow up PR, to merge this one quickly

File: test/new-e2e/tests/cws/fargate_test.go [hunk 3]
```diff
@@ -280,6 +280,11 @@ func getAgentFullImagePath(e *configCommon.CommonEnvironment) string {
 	if fullImagePath := e.AgentFullImagePath(); fullImagePath != "" {
 		return fullImagePath
 	}
+
+	if e.PipelineID() != "" && e.CommitSHA() != "" {
+		return fmt.Sprintf("669783387624.dkr.ecr.us-east-1.amazonaws.com/agent:%s-%s", e.PipelineID(), e.CommitSHA())
+	}
+
 	return agentDefaultImagePath
```
Comment (pducolin) [inline test/new-e2e/tests/cws/fargate_test.go:288]: **💭 thought**
We should probably expose this method on test-infra-side to re-use it

**Comment by pducolin** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=auto

**Review by ahmed-mez — APPROVED** [review]

**Comment by ahmed-mez** [comment]
@pducolin I believe this PR will fix the failure in the APM test https://github.com/DataDog/datadog-agent/pull/22787

File: test/new-e2e/tests/cws/fargate_test.go (hunk 3 above)
Comment (KevinFairise2) [reply test/new-e2e/tests/cws/fargate_test.go:288]: I think that's a good idea, since this on is just a copy/paste of the one we have on `test-infra`

**Comment by KevinFairise2** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=auto

**Review by YoannGh — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**